enhancements/okd/okd-featureset.md

```diff
@@ -0,0 +1,638 @@
+- Immutability validation (rejecting changes from OKD to other values)
+- Platform detection logic (OKD vs OpenShift)
+- CRD manifest generation for OKD variants
+
+**Integration Tests:**
+- API server correctly rejects attempts to enable OKD on OpenShift clusters
+- API server correctly rejects attempts to change from OKD to other feature sets
+- FeatureGate custom resource can be created with OKD feature set on OKD clusters
+- Feature gates are correctly applied when OKD feature set is enabled
+
+**E2E Tests:**
+- OKD cluster installs successfully with OKD feature set enabled by default
+- OKD cluster can be upgraded with OKD feature set enabled
+- Features enabled by the OKD feature set function correctly
+- OpenShift cluster installation/configuration rejects OKD feature set
+
+**Upgrade Tests:**
+- OKD clusters with OKD feature set can upgrade from version N to N+1
+- Feature set remains OKD after upgrade
+- Feature gates are correctly maintained across upgrades
+
+## Graduation Criteria
+
+The OKD feature set will be introduced as a stable feature, not following the typical Dev Preview -> Tech Preview -> GA progression, because:
+
+1. It is part of the OpenShift API contract
+2. The feature set mechanism is already GA
+3. This is adding a new value to an existing, stable enum
+4. OKD is already a mature distribution
+
+### Initial Release
+
+The OKD feature set will be considered stable when:
+
+- [ ] All API changes are merged to openshift/api
+- [ ] CRD manifests are generated for all relevant API resources
+- [ ] Validation logic is implemented and tested
+- [ ] Installer changes to enable OKD feature set by default are implemented
+- [ ] CI jobs for OKD with the new feature set are passing
+- [ ] Documentation is updated to describe the OKD feature set
+
+### Ongoing Requirements
+
+- Maintain compatibility with feature gate framework changes
+- Keep CRD manifests in sync as new API versions are added
+- Document any new features added to the OKD feature set
+- Ensure CI continues to test OKD with the feature set enabled
+
+## Upgrade / Downgrade Strategy
+
+### Upgrade Strategy
+
+**OKD Clusters:**
+- Existing OKD clusters without the OKD feature set: During the upgrade to the first version supporting the OKD feature set, the feature set should be automatically enabled if the cluster is detected as OKD. This should be handled by the cluster-version-operator or similar component.
+- OKD clusters with OKD feature set already enabled: No changes needed; the feature set persists across upgrades.
+- Upgrades are explicitly supported with the OKD feature set enabled.
+
+**OpenShift Clusters:**
+- No changes; the OKD feature set cannot be enabled and will not affect OpenShift clusters.
+
+### Downgrade Strategy
+
+**Downgrading from a version with OKD feature set to a version without:**
+- If an OKD cluster with the OKD feature set is downgraded to a version that does not recognize the OKD feature set:
```

Comment:
we should not classify this as supported - this will be a forced upgrade and may break clusters